src/bootstrap/lib.rs

```diff
@@ -65,9 +65,15 @@
 //! also check out the `src/bootstrap/README.md` file for more information.
 
 #![deny(warnings)]
+#![feature(associated_consts)]
+#![feature(core_intrinsics)]
```

Comment:
To be clear, the final patch landed without using the core intrinsic; we use normal, stable Any with `'static` types. In fact, rustbuild uses no unstable features (the only feature gate being the recently-stabilized associated constants).